vllm/v1/engine/async_llm.py

```diff
@@ -106,7 +106,11 @@ def __init__(
             executor_class=executor_class,
             log_stats=self.log_stats,
         )
-
+        for stat_logger in self.stat_loggers[0]:
+            if isinstance(stat_logger, PrometheusStatLogger) and isinstance(
+                    self.engine_core, MPClient):
```

Comment:
I'd remove this conditional - you have added `log_engine_initialized()` to the `StatLogger` base class and we should have the logger just access `VllmConfig` supplied to its constructor (rather than calling `get_gpu_blocks()`)